File: Java_App/src/main/java/com/example/acceltone/MenuPresenter.java
package com.example.acceltone;


import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Node;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.stage.Stage;

import java.io.IOException;

public class MenuPresenter {


    @FXML
    private Button accButton;
    @FXML
    private Button sigButton;
    private Stage stage;
    private Parent root;
    private Scene scene;

    @FXML
    public void initialize() {
        accButton.setOnAction(this::switchToAccelerator);
        sigButton.setOnAction(this::switchToSigGen);
    }

    @FXML
    private void switchToAccelerator(ActionEvent event) {
        try {
            root = FXMLLoader.load(getClass().getResource("accelerator.fxml"));
            stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
            scene = new Scene(root, Config.screenWidth, Config.screenHeight);
            stage.setScene(scene);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @FXML
    private void switchToSigGen(ActionEvent event) {
        try {
            root = FXMLLoader.load(getClass().getResource("signalgenerator.fxml"));
            stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
            scene = new Scene(root, Config.screenWidth, Config.screenHeight);
            stage.setScene(scene);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Scala_App/src/main/scala/MenuPresenter.scala

import javafx.event.ActionEvent
import javafx.fxml.{FXML, FXMLLoader}
import javafx.scene.control.Button
import javafx.scene.{Node, Parent, Scene}
import javafx.stage.Stage

import java.io.IOException


class MenuPresenter {
  @FXML var startButton: Button = _
  @FXML var stopButton: Button = _
  var screenHeight: Double = _
  var screenWidth: Double = _
  var stage: Stage = _
  var root: Parent = _
  var scene: Scene = _
  @FXML var accButton: Button = _

  @FXML var sigButton: Button = _


  @FXML
  def initialize(): Unit = {
    accButton.setOnAction((event: ActionEvent) => {
      switchToAccelerator(event)
    })
    sigButton.setOnAction((event: ActionEvent) => {
      switchToSigGen(event)
    })
  }

  @throws[IOException]
  def switchToAccelerator(event: ActionEvent): Unit = {

    printf("Screen resolution: %.0fx%.0f\n", screenWidth, screenHeight)
    root =  FXMLLoader.load(getClass.getResource("accelerator.fxml"))
    stage = event.getSource.asInstanceOf[Node].getScene.getWindow.asInstanceOf[Stage]
    scene = new Scene(root, Config.screenWidth, Config.screenHeight)
    stage.setScene(scene)
    stage.show()
  }

  @throws[IOException]
  def switchToSigGen(event: ActionEvent): Unit = {

    printf("Screen resolution: %.0fx%.0f\n", screenWidth, screenHeight)
    root = FXMLLoader.load(getClass.getResource("signalgenerator.fxml"))
    stage = event.getSource.asInstanceOf[Node].getScene.getWindow.asInstanceOf[Stage]
    scene = new Scene(root, Config.screenWidth, Config.screenHeight)
    stage.setScene(scene)
    stage.show()
  }

}